### ISSUE TITLE
lncli: `base_fee_msat` and `fee_rate` for `openchannel`

### ISSUE THREAD

**Issue by hieblmi** [lightningnetwork/lnd#6706]
### Background

Node operators often see their newly confirmed channels' local balance leaking to the remote side especially when connecting to capacity sinks while having configured low default fee rates. This issue requests the two parameters `base_fee_msat` and `fee_rate` for the `openchannel` option of `lncli` in order to default to the so specified fees immediately after a channel has been confirmed. This way immediate channel leakage can be avoided.

### Your environment

lnd v0.15.0-beta

### Expected behaviour

`lncli openchannel --node_key [pubkey] --local_amt [amt] --base_fee_msat 42 --fee_rate 1337` sets the appropriate fee policy after the channel has been confirmed.

**Comment by positiveblue** [comment]
I would recommend you to set really high fees by default until these fields are exposed in the `openchannel` API. That would help to ensure that the channel is not used until you set proper fees (or it would be profitable for you anyway).

You can change the defaults in the `lnd.conf`:

```
[Bitcoin]
// ...
bitcoin.basefee=5000000 # in msats
bitcoin.feerate=1000 # PPM in sats
```

**Comment by hieblmi** [comment]
@positiveblue - Would you mind if I would give it a shot? Wanted to use this issue to further familiarize myself with the lnd code. And thank you, yes - the default parameters in `lnd.conf` are a way around this but this is less convenient.

**Comment by hieblmi** [comment]
@positiveblue is fine with me taking a shot at this. Thanks!

**Comment by hieblmi** [comment]
Thanks @positiveblue for the pointers to delve into this issue. I want to summarize my findings so we can discuss the best approach to enable individual fee options for channel openings. 

Currently the pre-configured values `bitcoin.basefee` and `bitcoin.feerate` in `lnd.conf` are used to set the channel fees as part of the OpenChannel command in [funding/manager.go](https://github.com/lightningnetwork/lnd/blob/6980a18ab1d1f2affddb48636eae8fb0d33c83b4/funding/manager.go#L3507) when a [newChannelAnnouncement message](https://github.com/lightningnetwork/lnd/blob/6980a18ab1d1f2affddb48636eae8fb0d33c83b4/funding/manager.go#L3437) is sent after the 'funding locked' state is finalized and 6 block confirmations have passed.

The goal I assume is to add fee parameters to the [OpenChannelRequest](https://github.com/lightningnetwork/lnd/blob/d9c79d874e44770daf097042b4ca20a7c6e34008/cmd/lncli/cmd_open_channel.go#L252) and propagate these fields to the funding manager who can then check if individual fees should be used or whether the default fees from `lnd.conf` apply and should be used to create the channel announcement.

As far as I can tell there is currently no link to pass fee params between the client api and the [newChannelAnnouncement](https://github.com/lightningnetwork/lnd/blob/6980a18ab1d1f2affddb48636eae8fb0d33c83b4/funding/manager.go#L2921) call.

I potentially found one candidate path through the code to propagate the fields but I am unsure if this is the best way to do it so please advise here.

We could:

1. Add the fee options to the client message in [OpenChannelRequest](https://github.com/lightningnetwork/lnd/blob/d9c79d874e44770daf097042b4ca20a7c6e34008/lnrpc/lightning.pb.go#L6878)
2. Parse the request into a [InitFundingMessage](https://github.com/lightningnetwork/lnd/blob/d9c79d874e44770daf097042b4ca20a7c6e34008/funding/manager.go#L183) and pass it to the [server](https://github.com/lightningnetwork/lnd/blob/d9c79d874e44770daf097042b4ca20a7c6e34008/rpcserver.go#L2147)
3. Store the fee fields from the InitFundingMessage to [reservationWithContext](https://github.com/lightningnetwork/lnd/blob/6980a18ab1d1f2affddb48636eae8fb0d33c83b4/funding/manager.go#L3946) within the [handleInitFundingMsg](https://github.com/lightningnetwork/lnd/blob/6980a18ab1d1f2affddb48636eae8fb0d33c83b4/funding/manager.go#L3730)
4. Retrieve the reservation context and store the fee fields in the [completeChan here](https://github.com/lightningnetwork/lnd/blob/6980a18ab1d1f2affddb48636eae8fb0d33c83b4/funding/manager.go#L2085). With the limited context I have this step doesn't feel right to me as we'd have to add the fee fields to `*channeldb.OpenChannel`.
5. The fee fields could then be passed into the [ channel announcement ](https://github.com/lightningnetwork/lnd/blob/6980a18ab1d1f2affddb48636eae8fb0d33c83b4/funding/manager.go#L3122) and set for the [ChannelUpdate](https://github.com/lightningnetwork/lnd/blob/6980a18ab1d1f2affddb48636eae8fb0d33c83b4/funding/manager.go#L3507)

The End(afaik)

It would be great if someone can comment on the approach and provide guidance on how to properly implement channel fees for the OpenChannel command. 

Thank you.

**Comment by hieblmi** [comment]
@positiveblue, it would be great if you could review the linked PR.

**Comment by hieblmi** [comment]
@positiveblue I've added functionality so that we store the channel fees in case of a shutdown while we are waiting for a channel confirmation. Would you be able to review https://github.com/lightningnetwork/lnd/pull/6753 again?

**Comment by positiveblue** [comment]
Solved in #6753
